tests/telemetry/test_track.py

```diff
@@ -47,7 +45,7 @@ def fail_func():
     with pytest.raises(ValueError, match="test"):
         fail_func()
 
-    wait_for_telemetry_threads()
+    get_telemetry_client()._wait_for_consumer_threads()
```

Comment:
Can we kill the threads in the global telemetry client here?

Reply:
I see, then can we use `flush` here?

Reply:
Yea we should use flush :)